plugins/defaultplugins/ifplugin/stn_config.go

```diff
@@ -0,0 +1,170 @@
+func (plugin *StnConfigurator) Init() (err error) {
+	plugin.Log.Debug("Initializing StnConfigurator")
+
+	// Init VPP API channel
+	plugin.vppChan, err = plugin.GoVppmux.NewAPIChannel()
+	if err != nil {
+		return err
+	}
+
+	return plugin.checkMsgCompatibility()
+}
+
+// CheckStn rule raw data
+func CheckStn(stnInput *model_stn.StnRule, index ifaceidx.SwIfIndex, log logging.Logger) (*vppcalls.StnRule, error) {
+	if stnInput == nil {
+		return nil, fmt.Errorf("STN input is empty")
+	}
+	if stnInput.Interface == "" {
+		return nil, fmt.Errorf("STN input does not contain interface")
+	}
+	if stnInput.IpAddress == "" {
+		return nil, fmt.Errorf("STN input does not contain IP")
+	}
+
+	ifName := stnInput.Interface
+	ifIndex, _, exists := index.LookupIdx(ifName)
+	if !exists {
+		return nil, fmt.Errorf("STN entry interface %v not found", ifName)
+	}
+
+	parsedIP, _, err := addrs.ParseIPWithPrefix(stnInput.IpAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	stnRule := &vppcalls.StnRule{
+		IPAddress: *parsedIP,
+		IfaceIdx:  ifIndex,
+	}
+	return stnRule, nil
+}
+
+// Add create a new STN rule
+func (plugin *StnConfigurator) Add(rule *model_stn.StnRule) error {
+	plugin.Log.Infof("Creating new STN rule %v", rule)
+
+	// Check stn data
+	stnRule, err := CheckStn(rule, plugin.SwIfIndexes, plugin.Log)
+	if err != nil {
+		return err
+	}
+	plugin.Log.Debugf("adding STN rule: %+v", stnRule)
+	// Create and register new stn
+	errVppCall := vppcalls.AddStnRule(stnRule.IfaceIdx, &stnRule.IPAddress, plugin.Log, plugin.vppChan, measure.GetTimeLog(stn.StnAddDelRule{}, plugin.Stopwatch))
+	if errVppCall != nil {
+		return errVppCall
+	}
+	stnID := stnIdentifier(stnRule.IfaceIdx, stnRule.IPAddress.String())
+	plugin.StnIndexes.RegisterName(stnID, plugin.StnIndexSeq, nil)
+	plugin.StnIndexSeq++
+	plugin.Log.Infof("STN entry %v registered", stnID)
+
+	return nil
+}
+
+// Delete removes STN rule
+func (plugin *StnConfigurator) Delete(rule *model_stn.StnRule) error {
+	plugin.Log.Infof("Removing rule on if: %v with IP: %v", rule.Interface, rule.IpAddress)
+	// Check stn data
+	stnRule, err := CheckStn(rule, plugin.SwIfIndexes, plugin.Log)
+	if err != nil {
+		return err
+	}
+	if stnRule == nil {
+		return nil
+	}
+	plugin.Log.Debugf("deleting stn rule: %+v", stnRule)
+	// Remove and unregister route
+	err = vppcalls.DelStnRule(stnRule.IfaceIdx, &stnRule.IPAddress, plugin.Log, plugin.vppChan, measure.GetTimeLog(stn.StnAddDelRule{}, plugin.Stopwatch))
+	if err != nil {
+		return err
+	}
+	stnID := stnIdentifier(stnRule.IfaceIdx, stnRule.IPAddress.String())
+	_, _, found := plugin.StnIndexes.UnregisterName(stnID)
+	if found {
+		plugin.Log.Infof("STN rule %v unregistered", stnID)
+	} else {
+		plugin.Log.Warnf("Unregister failed, STN rule %v not found", stnID)
+	}
+
+	return nil
+}
+
+// Modify changes the stored rules
+func (plugin *StnConfigurator) Modify(rule *model_stn.StnRule, rule2 *model_stn.StnRule) error {
+	//TODO: Need to be implemented
```

Comment:
what about this? Are you going to implement it?

Reply:
Goal is to use it on next patch for STN rules where rules are modified after interface changes.